tests/run/test_genericclass.py

```diff
@@ -0,0 +1,135 @@
+# mode: run
+# tag: pure3.7
```

Comment:
Looks like I'm failing CI from this test on python versions <3.7. Any idea what might be wrong here @da-woods?

Reply:
In Py2, the classes are old-style classes, not types. Let them inherit from `(object)`.

And the `_patched` test depends on the `__init_subclass__` protocol, which is Py3.6+, it seems. You can disable it in Py<3.5 with a `@skipIf()` decorator.

Reply:
I was actually under the impression that the "tag" comment would instruct the test runner to ignore this test for python < 3.7 which is what I really wanted to do.

Reply:
"pure3.7" runs the (Cython compiled) tests in all versions of Python, and also runs the tests in uncompiled-Python for Python>3.7. I think this is correct because it should be possible for Cython to support `__class_getitem__` all the time. Essentially it's just providing confirmation that Cython matches Python behaviour (for the versions of Python where this is supported)

The `__init_subclass__` is probably something that has to be in the Python interpreter and Cython can't do on its own, so it makes sense to disable conditionally for some versions of Python.

Reply:
The old style classes are actually a bug in current master. With `language_level=3`, there shouldn't be any old style classes in Py2 any more. I pushed a fix as #3530, let's see if that breaks anything in the test suite.

Reply:
> The `__init_subclass__` is probably something that has to be in the Python interpreter and Cython can't do on its own

There's actually #2781 about this. PEP-487 Just needs to get implemented.